documentation/docs/10-adapters.md

```diff
@@ -2,45 +2,98 @@
+With this, [svelte-kit build](#command-line-interface-svelte-kit-build) will generate a self-contained Node app inside the `build` directory. You can pass options to adapters, such as customising the output directory:
 
 ```diff
 // svelte.config.js
-import node from '@sveltejs/adapter-node';
+import adapter from '@sveltejs/adapter-node';
 
 export default {
 	kit: {
--		adapter: node()
-+		adapter: node({ out: 'my-output-directory' })
+-		adapter: adapter()
++		adapter: adapter({ out: 'my-output-directory' })
 	}
 };
 ```
 
-A variety of official adapters exist for serverless platforms...
+### Creating a static site
+
+Most adapters will generate static HTML for any [prerenderable](#ssr-and-javascript-prerender) pages of your site. In some cases, your entire app might be prerenderable, in which case you can use `@sveltejs/adapter-static@next` to generate static HTML for _all_ your pages. A fully static site can be hosted on a wide variety of platforms, including static hosts like [GitHub Pages](https://pages.github.com/).
+
+```diff
+// svelte.config.js
+-import adapter from '@sveltejs/adapter-auto';
++import adapter from '@sveltejs/adapter-static';
+```
+
+You can also use `adapter-static` to generate single-page apps (SPAs) by specifying a [fallback page](https://github.com/sveltejs/kit/tree/master/packages/adapter-static#spa-mode).
+
+### Writing custom adapters
+
+We recommend [looking at the source for an adapter](https://github.com/sveltejs/kit/tree/master/packages) to a platform similar to yours and copying it as a starting point.
+
+Adapters packages must implement the following API, which creates an `Adapter`:
+
+```js
+/** @param {AdapterSpecificOptions} options */
+export default function (options) {
+	/** @type {import('@sveltejs/kit').Adapter} */
+	return {
+		name: 'adapter-package-name',
+		async adapt({ utils, config }) {
+			// adapter implementation
+		}
+	};
+}
+```
+
+The types for `Adapter` and its parameters are available in [types/config.d.ts](https://github.com/sveltejs/kit/blob/master/packages/kit/types/config.d.ts).
 
-- [`adapter-cloudflare`](https://github.com/sveltejs/kit/tree/master/packages/adapter-cloudflare) — for [Cloudflare Pages](https://developers.cloudflare.com/pages/)
-- [`adapter-cloudflare-workers`](https://github.com/sveltejs/kit/tree/master/packages/adapter-cloudflare-workers) — for [Cloudflare Workers](https://developers.cloudflare.com/workers/)
-- [`adapter-netlify`](https://github.com/sveltejs/kit/tree/master/packages/adapter-netlify) — for [Netlify](https://netlify.com)
-- [`adapter-vercel`](https://github.com/sveltejs/kit/tree/master/packages/adapter-vercel) — for [Vercel](https://vercel.com)
+Within the `adapt` method, there are a number of things that an adapter should do:
 
-...and traditional platforms:
+- Clear out the build directory
+- Output code that:
+  - Calls `init`
```

Comment:
Can the `init` being referred to here be clarified?

Similarly, can it be clarified where the `render` that's supposed to be called below comes from?

Or do you think that the current instructions will suffice if someone is simultaneously following along reading an existing adapter?

Reply:
yeah these docs were not all that complete. fleshing them out now